demofile-parsing/parse_demo.js

```diff
@@ -1,24 +1,119 @@
 const fs = require("fs");
 const demofile = require("demofile");
 const path = require("path");
+const Round = require('./Round.js');
 
-fs.readFile(path.resolve("/Users/tylerlam/Downloads/test.dem"), (err, buffer) => {
+const DEBUG = false;
+const demPath = "/Users/tylerlam/Desktop/demfiles/ence-vs-astralis-m2-inferno.dem";
+const streamGameStart = 595; // specific to the stream chosen. Must set this before running
+var demoGameStart = null;
+
+// maps seconds time of demofile to timestamp of video
+function secondsToTimestamp(seconds, demoStart, streamStart) {
+  demoStart = demoStart || 0;
+  seconds = seconds - demoStart + streamStart;
+  var hours = Math.floor(seconds/3600);
+  seconds = seconds%3600;
+  var mins = Math.floor(seconds/60);
+  seconds = Math.floor(seconds%60);
+  hours = hours < 10 ? `0${hours}` : hours;
+  mins = mins < 10 ? `0${mins}` : mins;
+  seconds = seconds < 10 ? `0${seconds}` : seconds;
+  return `${hours}h${mins}m${seconds}s`;
+}
+
+
+fs.readFile(path.resolve(b), (err, buffer) => {
   const demoFile = new demofile.DemoFile();
+  const rounds = [];
+  var roundEvents = [];
+  var roundLatestStart = 0;
+  var roundOn = false;
+  var roundIndex = 0;
+
+  // parse start of the round
+  demoFile.gameEvents.on("round_start", e => {
+    roundOn = true;
+    roundEvents = [];
+    var t = demoFile.currentTime;
+    roundLatestStart = t;
+    // log round start
+    console.log(`round ${roundIndex} started ${secondsToTimestamp(t, demoGameStart, streamGameStart)}`);
+
+    if (DEBUG) {
+      console.log(t);
+    }
+  });
+
+  // parse end of the round
+  demoFile.gameEvents.on("round_end", e => {
+    if(roundOn == true){
+      roundOn = false;
+      var t = demoFile.currentTime;
+      var round = new Round(roundLatestStart, t, roundEvents);
+      rounds.push(round);
+      // set the demo file game start to start time of round 1
+      demoGameStart = demoGameStart == null? roundLatestStart : demoGameStart;
+      // generate highlights for the round
+      round.sortKeyEvents();
+      round.calculateEventRates();
+      round.getHighRateTimes();
+      round.mergeTimeRanges();
+      round.mapToStreamTimestamps(demoGameStart, streamGameStart);
+      // reset roundEvents 
+      roundEvents = [];
+      // log round end
+      console.log(`round ${roundIndex} ended ${secondsToTimestamp(t, demoGameStart, streamGameStart)}`);
+      if (DEBUG) {
+        round.plotAllEvents();
+        round.plotHighRates();
+        round.plotHighlightTimes();
```

Comment:
Getting timestamps but not any ASCII Graph plotted. @mattlianje You get anything?

Reply:
Nevermind I'm a dumbo, it seems DEBUG isn't ever returned as true. Should we be keeping the "DEBUG" ifs?

Reply:
Set `DEBUG = true` on line 6 and you should see some graphy bois pop up